PokeDemo/Presenter/HomeViewPresenter.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  HomeViewPresenter.swift
+//  PokeDemo
+//
+//  Created by Rudr Bansal on 15/07/22.
+//
+
+import Foundation
+import UIKit
+
+protocol HomeViewPresenterDelegate: AnyObject {
+    func showPokemonList(pokemons: [Pokemon])
+    func showAlert(title: String, message: String)
+}
+
+typealias HomePresenterDelegate = HomeViewPresenterDelegate & UIViewController
+
+final class HomeViewPresenter {
+    
+    // MARK: Properties
+    
+    // MARK: Public
+    
+    weak var delegate: HomePresenterDelegate?
+    private let errorTitle = "Error"
+
+    // MARK: Private
+    
+    private let service: ServiceRepresentable
+    private var nextURL: String?
+
+    // MARK: Initailizers
+    
+    init(service: ServiceRepresentable = Service.shared) {
+        self.service = service
+    }
+    
+    // MARK: Exposed method(s)
+    
+    func setViewDelegate(delegate: HomePresenterDelegate) {
+        self.delegate = delegate
+    }
+    
+    func isLoadingCell(for indexPath: IndexPath) -> Bool {
```

Comment:
what is it used for?

Reply:
It was used to check if api should get more results.

Reply:
> It was used to check if api should get more results. Removed for now.

Where did you do it?

Reply:
cd97b93468f78a9f35b47f3476ee786dce2aca8a